src/main/java/org/opentripplanner/graph_builder/module/GtfsModule.java

```diff
@@ -181,6 +227,38 @@ private GtfsMutableRelationalDao loadBundle(GtfsBundle gtfsBundle)
 
         GtfsFeedId gtfsFeedId = gtfsBundle.getFeedId();
 
+        // check for duplicate bundle names and make sure that at most one of those bundles contains an auto-generated
+        // feed ID. If there is more than one, non-deterministic Feed IDs could be generated.
+        String bundleFilename = gtfsBundle.getPath().getName();
+        if (bundleFilenamesWithAutoGeneratedFeedId.contains(bundleFilename)) {
+            throw new RuntimeException(
+                String.format(
+                    "There are at least 2 bundles with the name `%s` that lack a feed_id value in a record in the feed_info.txt table. This will result in non-deterministic Feed ID generation!",
+                    bundleFilename
+                )
+            );
+        }
+        if (gtfsFeedId.isAutoGenerated()) {
+            bundleFilenamesWithAutoGeneratedFeedId.add(bundleFilename);
+            autoGeneratedFeedIds.add(gtfsFeedId.getId());
+        }
+
+        // read the agency table to determine if there is a single agency entry without an agency ID. This is used later
+        // to determine whether an agency ID was auto-generated.
+        boolean agencyFileContainedOneAgencyWithoutId = false;
+        InputStream agencyInputStream = gtfsBundle.getCsvInputSource().getResource("agency.txt");
+        try {
+            CsvReader result = new CsvReader(agencyInputStream, StandardCharsets.UTF_8);
+            result.readHeaders();
+            result.readRecord();
+            String firstAgencyId = result.get("agency_id");
```

Comment:
Reading the first agency record is not consistent with the comment above. Which one is right?

Reply:
I added some clarification comments.